packages/cloudflare/src/api/cloudflare-context.ts

```diff
@@ -0,0 +1,149 @@
+declare global {
+  interface CloudflareEnv {
+    NEXT_CACHE_WORKERS_KV?: KVNamespace;
+    ASSETS?: Fetcher;
+  }
+}
+
+export type CloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+> = {
+  /**
+   * the worker's [bindings](https://developers.cloudflare.com/workers/runtime-apis/bindings/)
+   */
+  env: CloudflareEnv;
+  /**
+   * the request's [cf properties](https://developers.cloudflare.com/workers/runtime-apis/request/#the-cf-property-requestinitcfproperties)
+   */
+  cf: CfProperties | undefined;
+  /**
+   * the current [execution context](https://developers.cloudflare.com/workers/runtime-apis/context)
+   */
+  ctx: Context;
+};
+
+// Note: this symbol needs to be kept in sync with the one used in `src/cli/templates/worker.ts`
+const cloudflareContextSymbol = Symbol.for("__cloudflare-context__");
+
+/**
+ * Utility to get the current Cloudflare context
+ *
+ * @returns the cloudflare context
+ */
+export async function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(): Promise<CloudflareContext<CfProperties, Context>> {
+  const global = globalThis as unknown as {
```

Comment:
could we add the type to `declare global` higher up in this file instead?

Reply:
I am not sure if you can extend the global declaration with a value indexed by a symbol, a bunch of googling lets me think that it's not possible 😕

Reply:
Too bad :(

We could either switch to a string or left as is then

Reply:
Another option:

```ts
const mySymbol = Symbol.for('internal.fake');
type MyCustomGlobal = typeof globalThis & {
  [mySymbol]: { loader: true}
}

(globalThis as MyCustomGlobal)[mySymbol] = { loader: true};
```

Reply:
> We could either switch to a string

I could really prefer not to do that because if we were to use a string and add it to the global declaration then that would leak in the user types, so with autocompletion etc... they would see such a field in their global scope (besides the fact that they could easily access it)

Reply:
> Another option:
> 
> ```ts
> const mySymbol = Symbol.for('internal.fake');
> type MyCustomGlobal = typeof globalThis & {
>   [mySymbol]: { loader: true}
> }
> 
> (globalThis as MyCustomGlobal)[mySymbol] = { loader: true};
> ```

I'm happy to go with that if you prefer, but to me looks more verbose than the current one, and I don't think it makes things clearer (a type defined inside a function is not super clear/readable in my opinion):
![Screenshot 2025-01-20 at 10 54 09](https://github.com/user-attachments/assets/46ab25b0-9748-4d42-9af0-75edc9cb71a2)

Reply:
Very good point for string.

The idea `typeof globalThis & {...}` would be to have this close to `declare global {...}` at the top of the file.
But then you would probably have a typing issue... which might be fixed with a function... so probably not worth it